src/administration/horizontal_scaling.md

```diff
@@ -0,0 +1,141 @@
+# Scaling Lemmy horizontally
+
+This is a collection of notes on scaling different Lemmy components horizontally. This is not meant as a step-by-step
+guide, rather as general tips and knowledge that might be useful to somebody who is already familiar with and horizontal scaling concepts and best
+practices. If you don't already know about concepts like load balancing,
+object storage, and reverse proxies, then this document will probably be hard (or impossible) to understand.
+
+**Note: scaling Lemmy this way is not necessary, and for small instances, the added overhead most likely
+outweighs any benefits.**
+
+## Why scale horizontally?
+
+- If one of your Lemmy servers dies for any reason, your instance can still remain operational
+- Having multiple Lemmy servers allows you to do rolling upgrades (no downtime needed to upgrade Lemmy!)
+- Horizontal scaling provides additional flexibility in how you upgrade your infrastructure
+- As of Lemmy version 0.18.2, Lemmy appears to be able to use the same resources more efficiently if they are split between multiple Lemmy processes
+
+## Breakdown of Lemmy components
+
+### Lemmy-ui
+
+Lemmy-ui is the main frontend for Lemmy. It consists of an expressjs based server-side process (necessary for SSR) and
+client code which run in the browser. It does not use a lot of resources and will happily run on quite low powered
+servers.
```

Comment:
This general breakdown of components could fit well into the main administration page. No need to hide it here as its also useful for admins who dont do horizontal scaling.